internal/serdeutil/raw_message.go

```diff
@@ -0,0 +1,31 @@
+package serdeutil
+
+import "github.com/pkg/errors"
+
+// RawMessage is a raw encoded JSON or YAML value.
+// It implements:
+// - [json.Marshaler] and [json.Unmarshaler]
+// - [yaml.BytesMarshaler] and [yaml.BytesUnmarshaler]
+// It can be used to delay JSON/YAML decoding or precompute a JSON/YAML encoding.
+type RawMessage []byte
+
+func (m RawMessage) MarshalJSON() ([]byte, error) { return m.marshal() }
+func (m RawMessage) MarshalYAML() ([]byte, error) { return m.marshal() }
+
+func (m RawMessage) marshal() ([]byte, error) {
+	if m == nil {
+		return []byte("null"), nil
+	}
+	return m, nil
+}
+
+func (m *RawMessage) UnmarshalJSON(data []byte) error { return m.unmarshal(data) }
```

Comment:
Go Docs recommends using consistent receivers definition, it there any reason you have used value in marshal operation?

Reply:
it's a limitation of json encoding library, if I have a struct which is included as a field value of another struct and I don't want it to be a pointer It won't get marshaled, see https://github.com/golang/go/issues/22967 
The ongoing discussion on json v2 library also describes this behaviour: https://github.com/golang/go/discussions/63397